packages/opentelemetry-api/src/internal/semver.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { VERSION } from '../version';
+
+const re = /^(\d+)\.(\d+)\.(\d+)(?:-(.*))?$/;
+
+/**
+ * Create a function to test an API version to see if it is compatible with the provided ownVersion.
+ *
+ * The returned function has the following semantics:
+ * - Exact match is always compatible
+ * - Major versions must always match
+ * - The minor version of the API module requesting access to the global API must be greater or equal to the minor version of this API
+ * - Patch and build tag differences are not considered at this time
+ *
+ * @param ownVersion version which should be checked against
+ */
+export function _makeCompatibilityCheck(
+  ownVersion: string
+): (version: string) => boolean {
+  const acceptedVersions = new Set<string>([ownVersion]);
+  const rejectedVersions = new Set<string>();
+
+  const myVersionMatch = ownVersion.match(re);
+  if (!myVersionMatch) {
+    throw new Error('Cannot parse own version');
```

Comment:
I think that the whole function should work as typical boolean check and in this case it should return false and maybe just log warning. Otherwise it might stop / break the user system

Reply:
If it can't parse it's own version that is something that will fail every single time and nothing the user can do about it. In this case I think throwing is the appropriate response since it's a bug that must be fixed. The throw should make this trip in tests and never allow us to release with a version which can't be parsed.